tests/display/test_inputs.py

```diff
@@ -41,6 +42,16 @@ def test_init_from_param(self, qtbot, default_value):
         assert bool_input.text() == bool_param.name
         assert bool_input.value() == default_value
 
+        # Second test is with value = not default_value
```

Comment:
Thank you for adding/adapting tests for this!

(This applies to all the tests changed/added here)
You are basically duplicating the test code here, as your comment also hints at a little. In general it's advisable to limit the number of asserts and different logic/configurations in any given test. This limits the opportunities for a test to break, which eases diagnosis _if_ the test fails.

Two approaches to improve the situation are:
* To split this into two tests with different names to indicate the two different "purposes" for the tests. This can duplicate test code of course, which is typically not very desirable but can be worth it if the differences between the test implementations are larger.
* Another way, especially if the code of the two different alternatives is nearly identical, is to stack another `parametrize` along the different option set you choose (see the end of [this doc section](https://docs.pytest.org/en/stable/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions)), which can be an elegant way to generate lots of combinations.

In this case, I think this should do the trick (decorating the test function, not the class, so only this test is affected)
```
@pytest.mark.parametrize("default_value", [True, False])
class TestBooleanInput:
    @pytest.mark.parametrize("value_remains_default", [True, False])
    def test_init_from_param(self, qtbot, default_value, value_remains_default):
``` 
then in the test do something like
```
if value_remains_default:
    ref = bool_param.default
else:
    ref = not bool_param.default
```
This should generate 4 tests with the approprite combinations without any duplicated test code. :-)

Reply:
Thanks you for the hints. I think what you proposed is very elegant and make a lot of sense. I have added in the update of the pull request.